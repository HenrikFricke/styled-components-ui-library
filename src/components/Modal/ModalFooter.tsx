import * as React from 'react';
import styled from 'styled-components';

import { PrimaryButton, SecondaryButton } from '../Button/Button';

interface ModalFooterProps {
    className?: string;
}

const ModalFooter: React.StatelessComponent<ModalFooterProps> = props => {
    return (
        <div className={props.className}>
            <PrimaryButton>Save</PrimaryButton>
            <SecondaryButton>Cancel</SecondaryButton>
        </div>
    );
};

export default styled(ModalFooter) `
    display: flex;
    flex-direction: row-reverse;

    > ${PrimaryButton} {
        margin-left: 10px;
    }
`;