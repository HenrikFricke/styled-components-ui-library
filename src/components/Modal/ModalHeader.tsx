import * as React from 'react';
import styled from 'styled-components';

import { H1 } from '../Headline/Headline';

import {
    colors
} from '../../constants';

interface ModalHeaderProps {
    className?: string;
    title: string;
}

const ModalHeader: React.StatelessComponent<ModalHeaderProps> = props => {
    return (
        <div className={props.className}>
            <H1>{props.title}</H1>
        </div>
    );
};

export default styled(ModalHeader) `
    border-bottom: 1px solid ${colors.PRIMARY};
`;