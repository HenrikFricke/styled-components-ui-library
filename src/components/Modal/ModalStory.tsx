import * as React from 'react';
import { storiesOf } from '@storybook/react';

import Modal from './Modal';

storiesOf('Modal', module)
    .add('default', () => (
        <Modal
            title="Dialog"
            text={`Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque a maximus risus. 
                Mauris pulvinar tellus vitae purus mollis, in finibus sapien eleifend.`}
            saveButtonClickHandler={() => null}
        />
    ));