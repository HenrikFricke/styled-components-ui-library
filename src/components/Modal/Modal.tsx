import * as React from 'react';
import styled from 'styled-components';

import ModalHeader from './ModalHeader';
import ModalFooter from './ModalFooter';
import Paragraph from '../Paragraph/Paragraph';

import {
    colors
} from '../../constants';

interface ModalProps {
    className?: string;
    title: string;
    text: string;
}

const Modal: React.StatelessComponent<ModalProps> = props => {
    return (
        <div className={props.className}>
            <ModalHeader title={props.title} />
            <Paragraph>{props.text}</Paragraph>
            <ModalFooter />
        </div>
    );
};

export default styled(Modal) `
    border: 1px solid ${colors.GRAY};
    border-radius: 5px;
    box-shadow: 0 0 10px -2px ${colors.GRAY};
    width: 600px;
    padding: 20px;
    box-sizing: border-box;
    background-color: ${colors.WHITE};
`;