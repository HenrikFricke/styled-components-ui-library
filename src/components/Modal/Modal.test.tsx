import * as React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import 'jest-styled-components';

import Modal from './Modal';

describe('Modal', () => {
    let tree: ReactTestRenderer.ReactTestRendererJSON;

    beforeEach(() => {
        tree = ReactTestRenderer.create(
            <Modal
                title="Dialog"
                text={`Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque a maximus risus. 
                Mauris pulvinar tellus vitae purus mollis, in finibus sapien eleifend.`}
            />
        ).toJSON();
    });

    it('should render correctly', () => {
        expect(tree).toMatchStyledComponentsSnapshot();
    });
});