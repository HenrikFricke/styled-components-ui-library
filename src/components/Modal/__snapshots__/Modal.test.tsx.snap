// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
.bVyChL {
  color: palevioletred;
  padding: 10px 0;
  font-size: 32px;
  margin: 0;
}

.jteMLO {
  border-bottom: 1px solid palevioletred;
}

.jDjMbl {
  border: 2px solid palevioletred;
  background-color: palevioletred;
  color: white;
  padding: 10px;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: 300ms all;
  transition: 300ms all;
  font-size: 16px;
}

.jDjMbl:hover {
  background-color: mediumvioletred;
  border-color: mediumvioletred;
}

.jexVLz {
  border: 2px solid palevioletred;
  background-color: palevioletred;
  color: white;
  padding: 10px;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: 300ms all;
  transition: 300ms all;
  font-size: 16px;
  background-color: transparent;
  color: palevioletred;
}

.jexVLz:hover {
  background-color: mediumvioletred;
  border-color: mediumvioletred;
}

.jexVLz:hover {
  background-color: transparent;
  color: mediumvioletred;
}

.fMNUAa {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.fMNUAa > .sc-htpNat {
  margin-left: 10px;
}

.nABcD {
  color: black;
  padding: 5px;
  font-size: 16px;
  line-height: 1.5;
  -webkit-text-align: justify;
  text-align: justify;
}

.eOpneg {
  border: 1px solid gainsboro;
  border-radius: 5px;
  box-shadow: 0 0 10px -2px gainsboro;
  width: 600px;
  padding: 20px;
  box-sizing: border-box;
  background-color: white;
}

<div
  className="sc-bZQynM eOpneg"
>
  <div
    className="sc-bwzfXH jteMLO"
  >
    <h1
      className="sc-bdVaJa bVyChL"
    >
      Dialog
    </h1>
  </div>
  <p
    className="sc-EHOje nABcD"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque a maximus risus. 
                    Mauris pulvinar tellus vitae purus mollis, in finibus sapien eleifend.
  </p>
  <div
    className="sc-ifAKCX fMNUAa"
  >
    <button
      className="sc-htpNat jDjMbl"
    >
      Save
    </button>
    <button
      className="sc-bxivhb jexVLz"
    >
      Cancel
    </button>
  </div>
</div>
`;
